eis_toolkit/exploratory_analyses/dbscan.py

```diff
@@ -0,0 +1,62 @@
+import geopandas as gdp
+import numpy as np
+from beartype import beartype
+from sklearn.cluster import DBSCAN
+
+from eis_toolkit.exceptions import EmptyDataFrameException, InvalidParameterValueException
+
+
+def _dbscan(data: gdp.GeoDataFrame, max_distance: float, min_samples: int) -> gdp.GeoDataFrame:
+
+    coordinates = list(data.geometry.apply(lambda geom: [geom.x, geom.y]))
+    dbscan = DBSCAN(eps=max_distance, min_samples=min_samples)
+
+    dbscan.fit(coordinates)
+    data["cluster"] = dbscan.labels_
+
+    core_indices = dbscan.core_sample_indices_
+    mask = np.zeros(len(coordinates), dtype=int)
+    mask[core_indices] = 1
+    data["core"] = mask
+
+    return data
+
+
+@beartype
+def dbscan(data: gdp.GeoDataFrame, max_distance: float = 0.5, min_samples: int = 5) -> gdp.GeoDataFrame:
+    """
+    Perform DBSCAN clustering on the input data.
+
+    Args:
+        data: GeoDataFrame containing the input data.
+        max_distance: The maximum distance between two samples for one to be considered as in the neighborhood of
+            the other. Optional parameter.
+        min_samples: The number of samples in a neighborhood for a point to be considered as a core point.
+            Optional parameter.
+
+    Returns:
+        GeoDataFrame containing assigned cluster labels and value 1 if a data point is a core point
+        and 0 otherwise.
```

Comment:
I would make this description slightly clearer like this:
```suggestion
        GeoDataFrame containing two new columns: one with assigned cluster labels and one indicating whether a point is a core point (1) or not (0).
```